### PR TITLE
[ci] Flamegraph and stack perf data

### PR DESCRIPTION
**Pull request by anandbonde**
Creates flamegraph and stack perf data for PRs.

Flamegraphs (in SVG format) will be stored in archives. Stack perf data will be archived as well as posted as comments on PRs.

### CODE REVIEW

File: src/rust/catnap/linux/transport.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ use crate::{
         DemiRuntime,
         SharedDemiRuntime,
         SharedObject,
-    },
+    }, timer,
```
Comment (ppenna) [inline src/rust/catnap/linux/transport.rs:41]: Not sure if this was auto-formatted (if it was, just ignore).
To keep consistency, could you move this to the next line?

File: src/rust/catnap/win/overlapped.rs
Comment (ppenna) [inline src/rust/catnap/win/overlapped.rs:1]: Dumb question: what does IOC stand for?

File: src/rust/catnap/linux/transport.rs
Comment (ppenna) [inline src/rust/catnap/linux/transport.rs:1]: Dumb question: what does bgc stand for?

**Review by ppenna — APPROVED** [review]
LGTM

File: src/rust/catnap/linux/transport.rs (hunk 1 above)
Comment (anandbonde) [reply src/rust/catnap/linux/transport.rs:41]: Yes, I will move it. Thanks.

File: src/rust/catnap/win/overlapped.rs
Comment (anandbonde) [reply src/rust/catnap/win/overlapped.rs:1]: ioc -> io coroutine

File: src/rust/catnap/linux/transport.rs
Comment (anandbonde) [reply src/rust/catnap/linux/transport.rs:1]: bgc -> background coroutine (I abbreviated it so that the collapsed stack is not a very long string).

**Pull request merged**